### PR TITLE
Templating

### LINKED ISSUE #13: The bump_cwl_version.py utility CI script messes up the CWL YAML order
[Before](https://github.com/Sage-Bionetworks-Workflows/dockstore-tool-template/blob/b28a40471a9e41ee4fe3a3dd7cdb9f0448988d29/cwl/echo_best_practices.cwl) and [after](https://github.com/Sage-Bionetworks-Workflows/dockstore-tool-template/blob/8aaeef1d4649d9d36e95f865fe96040b0034c25d/cwl/echo_best_practices.cwl) the `ci.yaml` GitHub Actions workflow was run.

Here's why we should fix this:
- The `echo_best_practices.yaml` file is a good starting point when creating new tools. 
- The unsorted, collapsed output from the `bump_cwl_version.py` script makes the CWL files harder to read/maintain.
- The diffs for the CI commits are larger than they need to be. 

I don't know yet if this is a simple fix (_e.g._ by tweaking a function argument when using PyYAML) or a more substantial change (_e.g._ by parsing the CWL file without PyYAML).

### LINKED ISSUE #15: Look into templating for CWL edit
Currently CWL edit step uses a python script; considering using templating such as moustache, handlebars, etc.

### PR DESCRIPTION
**Pull request by thomasyu888**
had to start ANOTHER new PR because of github secrets issue...

fixes #15 , fixes #13

### CODE REVIEW

**Review by tthyer — APPROVED** [review]
@thomasyu888 This looks good to me -- I'm eager to get templating into the tool
@BrunoGrandePhD Please review

File: .github/workflows/ci.yaml
Comment (zaro0508) [inline .github/workflows/ci.yaml:50]: to make it more generic how about passing in a json `input` or `data` dictionary here.  something like..
```
{
  "data": {
    "version": "1.0.0"
  }
}
```

then you can reference it with `{{ data.version }}` in your mustache templates.  that way you can pass in more than one piece of data.

**Review by BrunoGrandePhD — CHANGES_REQUESTED** [review]
I was hoping to test the CI workflow, but I haven't gotten `act` to work locally and I don't want to hold up this PR for that. It'll become clear if it works or not once it's merged. 

All I request is that the `echo_best_practices.cwl` file be restored to what I committed to the repository before it was mangled. I'll approve once that's done.

File: Pipfile.lock
Comment (thomasyu888) [inline Pipfile.lock:448]:
```suggestion
        "more-itertools": {
```

File: Pipfile.lock
Comment (thomasyu888) [inline Pipfile.lock:502]:
```suggestion
        "wcwidth": {
```

**Review by BrunoGrandePhD — APPROVED** [review]
🚀

**Pull request merged**
